packages/typegpu/src/data/attributes.ts

```diff
@@ -337,6 +337,9 @@ class DecoratedImpl<TInner extends BaseWgslData, TAttribs extends unknown[]>
   implements Decorated<TInner, TAttribs>
 {
   public readonly type = 'decorated';
+  public readonly '~memIdent'!: TAttribs extends Location<number>[]
+    ? MemIdentity<TInner> | Decorated<MemIdentity<TInner>, TAttribs>
+    : Decorated<TInner, TAttribs>;
```

Comment:
I am not 100% sure, but wouldn't the MemIdentity of a Decorated type be this instead?:
```suggestion
  public readonly '~memIdent'!: TAttribs extends Location<number>[]
    ? MemIdentity<TInner> | Decorated<MemIdentity<TInner>, TAttribs>
    : Decorated<MemIdentity<TInner>, TAttribs>;
```